assets/custom.js

```diff
@@ -494,4 +491,146 @@ $('.header__menu-dropdown').hover(
     function(){
         $(this).addClass('display-none');
     }
-);
\ No newline at end of file
+);
+
+
+// Display cart side-drawer when user clicks cart icon
+$('.header__icon--cart').click(function(){
+    if ($('.mini-cart-modal').hasClass('display-none')) {
+        $('.mini-cart-modal').removeClass('display-none');
+        $('body').addClass('overflow-hidden');
+    } else {
+        $('.mini-cart-modal').addClass('display-none');
+        $('body').removeClass('overflow-hidden');
+    }
+});
+
+function displayMiniCart() {
+    console.log('displayMiniCart()')
+    $('.mini-cart-modal').removeClass('display-none');
+    $('body').addClass('overflow-hidden');
+}
+
+//Disables bubbling to modal content's parent elements for cart side drawer
+$('.mini-cart').click(function(e) {
+    e.stopPropagation();
+});
+
+// Hide cart side-drawer 
+$('.mini-cart__close').click(function(){
+    $('.mini-cart-modal').addClass('display-none');
+    $('body').removeClass('overflow-hidden');
+});
+$('.mini-cart-modal').click(function(){
+    $('.mini-cart-modal').addClass('display-none');
+    $('body').removeClass('overflow-hidden');
+});
```

Comment:
i think you can combine these two to one click handler: `$('.mini-cart__close, .mini-cart-modal').click(...`

Reply:
Done. Can't seem to be able to combine the 'X' and outer modal for the newsletter subscription (line 401) though. If I erase the function starting at line 408, it seems to break the code at 401(it won't 'console.log($(this))' when clicking on the outside modal).